docs/source/credentials.rst

```diff
@@ -0,0 +1,39 @@
+Credentials
+***********
+
+Integration with Git
+====================
+
+Git and DataLad can use each other's credential system.
+In order to allow Git to query credentials from DataLad, the git credential helper delivered with DataLad needs to be configured.
+That is, a section like this needs to be part of one's git config file::
+
+  [credential "https://*.data.example.com"]
+    helper = "datalad"
+
+Note:
+
+- This most likely only makes sense at the user or system level (options `--global`|`--system` with `git config`), since cloning of a repository needs the credentials before there is a local repository.
+- The name of that section is a URL matching expression - see `man gitcredentials`.
+- The URL matching does NOT include the scheme! Hence, if you need to match `http` as well as `https`, you need two such entries.
+- Multiple different git credential helpers can be configured - Git will ask them one after another. For example on OSX, Git comes with a helper to use the OSX keychain which is configured at system level. This does not conflict with setting up DataLad's credential helper.
```

Comment:
> "Git will ask them one after another": does this mean all will always be asked? Or will it query them starting with the first, and attempt them sequentially until one matches and succeeds?

This. Which relates to the suspicion of cycles. Will change wording accordingly.

> OSX -> macOS (they've dropped the OSX for years now, as they are now on version 11.x)

Well. But the helper I am referring to is still called `git-credential-osxkeychain`. Will try to reword whie aowing to identify the one I am talking about.

> "This does not conflict with setting up DataLad's credential helper." I don't quite know what this means (and confuses me more than it helps me). How would the keychain "conflict"? It's just yet another credential helper that can be invoked by DataLad/git, yes?

Yes. I thought it may be a natural question: "There already is a helper configured. Am I supposed to overwrite that config?", because in a lot of cases, a git config you set up, will take precedence over the system config. Not in this case: It's an addition.